utf8.c

```diff
@@ -3140,7 +3140,7 @@ Perl_utf8_to_bytes_(pTHX_ U8 **s_ptr, STRLEN *lenp, U8 ** free_me,
             SAVEFREEPV(*s_ptr);
         }
         else {
-            *free_me = *s_ptr;
+            *free_me = (void *) *s_ptr;
```

Comment:
I don't think this cast is needed.

Reply:
I thought it would be helpful to future readers to emphasize this.

Reply:
I was more confused by the cast.

I tend to think of casts as dangerous - they bypass the type checking done by the compiler, so an unnecessary cast makes me wonder what I'm missing.

Reply:
Ok. That's a reasonable position